rust/operator-binary/src/controller.rs

```diff
@@ -984,6 +1019,43 @@ fn validated_product_config(
         .context(InvalidProductConfigSnafu)
 }
 
+async fn create_shared_internal_secret(trino: &TrinoCluster, client: &Client) -> Result<()> {
+    let secret = build_shared_internal_secret(trino)?;
+    if !client
+        .exists::<Secret>(&secret.name(), secret.namespace().as_deref())
+        .await
+        .context(ApplyInternalSecretSnafu)?
+    {
+        client
+            .apply_patch(FIELD_MANAGER_SCOPE, &secret, &secret)
+            .await
+            .context(ApplyInternalSecretSnafu)?;
+    }
+
+    Ok(())
+}
+
+fn build_shared_internal_secret(trino: &TrinoCluster) -> Result<Secret> {
+    let mut internal_secret = BTreeMap::new();
+    internal_secret.insert(INTERNAL_SECRET.to_string(), Uuid::new_v4().to_string());
```

Comment:
Could you please switch from UUID to openssl rand 512 bytes so that we have the recommended level of security from the docs?
I should be pretty easy, e.g. something like https://docs.rs/openssl/0.10.40/openssl/rand/fn.rand_bytes.html

Reply:
done